WinImageBuilder.psm1

```diff
@@ -873,250 +947,32 @@ function Run-Sysprep {
-     unstable images.
-    .PARAMETER Type
-     This parameter allows to choose between MAAS, KVM and Hyper-V specific images.
-    .PARAMETER PurgeUpdates
-     If set to true, will run DISM with /resetbase option. This will reduce the size of 
-     WinSXS folder, but after that Windows updates cannot be uninstalled.
-    .PARAMETER DisableSwap
-     DisableSwap option will disable the swap when the image is generated and will add a setting
-     in the Unattend.xml file which will enable swap at boot time during specialize step. This
-     is required as by default, the amount of swap space on Windows machine is directly
-     proportional to the RAM size and if the image has in the initial stage low disk space,
-     the first boot will fail due to not enough disk space. The swap is set to the default
-     automatic setting right after the resize of the partitions is performed by cloudbase-init.
-    #>
     [CmdletBinding()]
     param
-    (
-        [parameter(Mandatory=$true, ValueFromPipeline=$true)]
-        [string]$WimFilePath = "D:\Sources\install.wim",
-        [parameter(Mandatory=$true)]
-        [string]$ImageName,
-        [parameter(Mandatory=$true)]
-        [string]$WindowsImagePath,
-        [parameter(Mandatory=$true)]
-        [Uint64]$SizeBytes,
-        [ValidateSet("BIOS", "UEFI", ignorecase=$false)]
-        [string]$DiskLayout = "BIOS",
-        [parameter(Mandatory=$false)]
-        [string]$ProductKey,
-        [parameter(Mandatory=$false)]
-        [string]$VirtIOISOPath,
-        [parameter(Mandatory=$false)]
-        [switch]$InstallUpdates,
-        [parameter(Mandatory=$false)]
-        [string]$AdministratorPassword = "Pa`$`$w0rd",
-        [array]$ExtraFeatures = @(),
-        [parameter(Mandatory=$false)]
-        [string]$ExtraDriversPath,
-        [parameter(Mandatory=$false)]
-        [switch]$PersistDriverInstall = $true,
-        [parameter(Mandatory=$false)]
-        [Uint64]$Memory=2GB,
-        [parameter(Mandatory=$false)]
-        [int]$CpuCores=1,
-        [parameter(Mandatory=$false)]
-        [switch]$RunSysprep=$true,
-        [parameter(Mandatory=$false)]
-        [string]$SwitchName,
-        [parameter(Mandatory=$false)]
-        [switch]$Force=$false,
-        [ValidateSet("MAAS", "KVM", "HYPER-V", ignorecase=$false)]
-        [string]$Type = "MAAS",
-        [parameter(Mandatory=$false)]
-        [switch]$PurgeUpdates,
-        [parameter(Mandatory=$false)]
-        [switch]$DisableSwap,
-        [parameter(Mandatory=$false)]
-        [string]$ZipPassword
+        [parameter(Mandatory=$false, ValueFromPipeline=$true)]
+        [string]$ConfigFilePath =""
     )
     PROCESS
     {
+        $CONFIG = Get-AvailableConfigs $ConfigFilePath
         Write-Host ("Windows online image generation started at: {0}" -f @(Get-Date))
         Is-Administrator
-        if (!$RunSysprep -and !$Force) {
+        if (!$CONFIG["run_sysprep"] -and !$CONFIG["force"]) {

```

Comment:
these values are currently all equivalent to true, if the change above with the isboolean flag is not made